falcon/asgi/app.py

```diff
@@ -521,11 +477,17 @@ async def __call__(self, scope, receive, send):  # noqa: C901
             })
 
             await send(_EVT_RESP_EOF)
-            self._schedule_callbacks(resp)
+
+            # PERF(vytas): Check resp._registered_callbacks directly to shave
+            #   off a function call since this is a hot/critical code path.
+            if resp._registered_callbacks:
```

Comment:
can't we cache `_registered_callbacks`? it's used multiple times so I think creating a local variable may be worth it

Reply:
It is never assumed to be hit in the same path. (I.e., we should not be calling `self._schedule_callbacks(...)` twice.)

Reply:
ok, then I guess if we replace the argument of `_schedule_callbacks` it may be useful, otherwise maybe just a readability improvement